aiomqtt/client.py

```diff
@@ -511,15 +511,19 @@ async def connect(self, *, timeout: float | None = None) -> None:
         if self._disconnected.done():
             self._disconnected = asyncio.Future()
 
-    async def disconnect(self, *, timeout: float | None = None) -> None:
-        """Disconnect from the broker."""
-        # Early out if already disconnected...
+    def _early_out_on_disconnected(self)-> bool:
+         # Early out if already disconnected...
         if self._disconnected.done():
-            disc_exc = self._disconnected.exception()
-            if disc_exc is not None:
+            if (disc_exc:= self._disconnected.exception()) is not None:
```

Comment:
Rewrite to avoid `:=` since that's a python 3.8 feature (and we still support 3.7).

...Good use case for `:=`, though. :)